### PR TITLE
[DEVHUB-1703] Implement remaining notifications.

### PR DESCRIPTION
**Pull request by mcolella14**
## Jira Ticket:

[DEVHUB-1703](https://jira.mongodb.org/browse/DEVHUB-1703)

## Description:

- Added a hook (`useUserPreferences`) to utilize the notification context whenever updating preferences.
- Added `failedToFetch` flag on the session object to indicate that we failed to find the user's information.
- Adjust `submitPersonalizationSelections` to throw errors consistently.
- Used these to implement notifications.
- Also adjusted the `Cache-Control` header on personalization endpoints to invalidate the browser cache when sending a bad response. I was having some issues with this and the SWR cache, but it seems to work now.

### CODE REVIEW

**Review by justinmckenzie — APPROVED** [review]
nice thinking putting everything in a hook

File: src/hooks/personalization/user-preferences.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                    variant: 'WARN',
+                });
+                return;
+            }
+            setNotification({
+                message: 'Successfully saved your preferences',
+                variant: 'SUCCESS',
+            });
+        },
+        []
```
Comment (max-melo) [inline src/hooks/personalization/user-preferences.ts:37]: is there a particular reason why `setNotification` isn't in the dependency array? if not, i think it's worth adding... not sure if react guarantees the context value is populated before every render, but i'm seeing a default `() => null` value for setNotification, so best to be safe imo

Comment (mcolella14) [reply src/hooks/personalization/user-preferences.ts:37]: Ahh yep I just missed that, will add that, thanks for the catch!

**Review by harikakotipalli — APPROVED** [review]

**Review by max-melo — APPROVED** [review]
LGTM!

**Pull request merged**